### PR TITLE
[Templates] Diagnostics improvements and certificate fixes

### PR DESCRIPTION
**Pull request by javiercn**
* Move template specific helpers back to templates.
* Fix casing issue when loading the certificate.
* Capture startup output and throw when the app can't find the listening URL

### CODE REVIEW

**Review by pranavkm — APPROVED** [review]
As long as it works 👍

File: src/ProjectTemplates/Shared/AspNetProcess.cs [hunk 1]
```diff
@@ -134,13 +136,13 @@ public async Task AssertPagesOk(IEnumerable<Page> pages)
...
-            var response = await RequestWithRetries(client =>
+            var response = await RetryHelper.RetryRequest(async () =>
             {
                 var request = new HttpRequestMessage(
                     HttpMethod.Get,
                     new Uri(ListeningUri, page.Url));
-                return client.SendAsync(request);
-            }, _httpClient);
+                return await _httpClient.SendAsync(request);
+            }, logger: NullLogger.Instance);
```
Comment (dougbu) [inline src/ProjectTemplates/Shared/AspNetProcess.cs:145]: Does this prevent logging of errors hit in the final attempt when that also fails?

File: src/ProjectTemplates/Shared/AspNetProcess.cs [hunk 2]
```diff
@@ -232,21 +234,27 @@ private Uri GetListeningUri(ITestOutputHelper output)
...
+                    }
+                }
             }
             catch (OperationCanceledException)
             {
-                return null;
             }
+
+            throw new InvalidOperationException(@$"Couldn't find listening url:
+{string.Join(Environment.NewLine, buffer)}");
```
Comment (dougbu) [inline src/ProjectTemplates/Shared/AspNetProcess.cs:257]: 👍

File: src/ProjectTemplates/Shared/AspNetProcess.cs [hunk 3]
```diff
@@ -260,14 +268,13 @@ public Task AssertOk(string requestUrl)
         public Task AssertNotFound(string requestUrl)
             => AssertStatusCode(requestUrl, HttpStatusCode.NotFound);
 
-        internal Task<HttpResponseMessage> SendRequest(string path)
-        {
-            return RequestWithRetries(client => client.GetAsync(new Uri(ListeningUri, path)), _httpClient);
-        }
+        internal Task<HttpResponseMessage> SendRequest(string path) =>
+            RetryHelper.RetryRequest(() => _httpClient.GetAsync(new Uri(ListeningUri, path)), logger: NullLogger.Instance);
```
Comment (dougbu) [inline src/ProjectTemplates/Shared/AspNetProcess.cs:272]: Another question about hiding details here&hellip;

File: src/ProjectTemplates/Shared/AspNetProcess.cs [hunk 4]
```diff
@@ -277,8 +284,9 @@ public async Task AssertStatusCode(string requestUrl, HttpStatusCode statusCode,
                     request.Headers.Add("Accept", acceptContentType);
                 }
 
-                return client.SendAsync(request);
-            }, _httpClient);
+                return _httpClient.SendAsync(request);
+            }, logger: NullLogger.Instance);
```
Comment (dougbu) [inline src/ProjectTemplates/Shared/AspNetProcess.cs:288]: &hellip;and here

**Review by dougbu — COMMENTED** [review]
LGTM though I have a couple of questions about hidden output / details

File: src/ProjectTemplates/Shared/AspNetProcess.cs (hunk 1 above)
Comment (javiercn) [reply src/ProjectTemplates/Shared/AspNetProcess.cs:145]: Nope, the error/exception will show up at a higher level. The other helper didn't have logging there, so we are not loosing anything here. Same for all the other changes.

**Comment by javiercn** [comment]
Thanks @dougbu for baby sitting

**Pull request merged**